src/server/render.js

```diff
@@ -184,8 +184,19 @@ export function createRenderFunction (
     }
   }
 
+  function hasAncestorData (node: VNode) {
+    const parentNode = node.parent
+    return parentNode && (parentNode.data || hasAncestorData(parentNode))
+  }
+
   function renderStartingTag (node: VNode) {
     let markup = `<${node.tag}`
+
+    // construct synthetic data for module processing
+    // because modules like style also produce code by parent VNode data
+    if (!node.data && hasAncestorData(node)) {
```

Comment:
Add a synthetic data property if VNode's ancestor has data. I don't this is good enough...